packages/gatsby-link/src/index.js

```diff
@@ -3,7 +3,7 @@ import { Link, NavLink } from "react-router-dom"
 import PropTypes from "prop-types"
 
 let pathPrefix = ``
-if (__PREFIX_PATHS__) {
+if (typeof __PREFIX_PATHS__ !== "undefined" && __PREFIX_PATHS__) {
```

Comment:
@Alaev lodash `isUndefined` can catch if a variable is declared and set to `undefined`, but it will throw a reference error like any other function call if it is passed a variable name that has not been declared at all.  The only things I know of that can handle completely undeclared variable names are `typeof` or a try-catch block.